### PR TITLE
REF: Split libs into separate packages

### PR DESCRIPTION
**Pull request by carterbox**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Split shared libs and static libs into separate packages, so dynamic links are used by default and run_exports are smaller.

Addresses part of #12

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,30 +23,89 @@ requirements:
...
+        - if not exist "%LIBRARY_BIN%\\libopenlibm.dll" exit 1         # [win]
+        - if not exist "%LIBRARY_LIB%\\libopenlibm.lib" exit 1         # [win]
+        - echo on  # [win]
+        - cl /Tc test_openlibm.c /I %CONDA_PREFIX%\Library\include\openlibm /link /LIBPATH:"%CONDA_PREFIX%\Library\lib" libopenlibm.lib /out:test_openlibm.exe  # [win]
+        - test_openlibm.exe  # [win]
+        - ${CC} -o test_openlibm test_openlibm.c -I${CONDA_PREFIX}/include/openlibm -L${CONDA_PREFIX}/lib -lopenlibm ${CFLAGS} ${LDFLAGS}  # [unix]
+        - ./test_openlibm  # [unix]
+
+  - name: libopenlibm{{ so_major_version }}  # [unix]
+  - name: libopenlibm                        # [win]
```
Comment (carterbox) [inline recipe/meta.yaml:67]: I think the linter is mistaken because of how I have defined the libopenlibm packages on windows vs unix.

**Review by ngam — APPROVED** [review]
Looks good. If desired, please feel free to add yourself as maintainer.

**Comment by ngam** [comment]
(Will wait a day or two before applying the automerge label)

**Comment by carterbox** [comment]
Thanks for the review! I have no desire to become a maintainer, I was just reviewing a PR over on staged-recipes that was/is statically linking to this package. This refactoring is so that they don't statically link.

**Pull request merged**